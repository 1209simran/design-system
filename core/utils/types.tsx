export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
