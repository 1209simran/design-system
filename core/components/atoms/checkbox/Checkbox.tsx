import * as React from 'react';
import classNames from 'classnames';
import Text from '@/components/atoms/text';
import Icon from '../icon';

export type Size = 'regular' | 'tiny';

export interface ICheckboxProps {
  size?: Size;
  checked?: boolean;
  intermediate?: boolean;
  disabled?: boolean;
  label?: string;
  name?: string;
  value?: string;
  onChange?: (checked: boolean) => void;
}

const Checkbox = React.forwardRef<HTMLInputElement, ICheckboxProps>((props, forwardedRef) => {
  const {
    size = 'regular',
    label,
    disabled,
    onChange,
    name,
    value,
  } = props;

  const ref = React.useRef<HTMLInputElement>(null);

  React.useImperativeHandle(forwardedRef, (): HTMLInputElement => {
    return ref.current as HTMLInputElement;
  });

  const [checked, setChecked] = React.useState(props.checked);

  React.useEffect(() => {
    setIntermediate(props.intermediate);
  }, [props.intermediate]);

  React.useEffect(() => {
    setChecked(props.checked);
  }, [props.checked]);

  const CheckboxClass = classNames({
    ['Checkbox']: true,
    ['Checkbox--disabled']: disabled,
    [`Checkbox--${size}`]: size,
  });

  const CheckboxWrapper = classNames({
    ['Checkbox-wrapper']: true,
    [`Checkbox-wrapper--${size}`]: size,
    ['Checkbox-wrapper--checked']: checked,
    ['Checkbox-wrapper--intermediate']: props.intermediate,
  });

  const setIntermediate = (intermediate: any) => {
    ref.current!.indeterminate = intermediate;
  };

  const onChangeHandler = () => {
    const checkedValue = (props.intermediate) ? true : !checked;
    setChecked(checkedValue);
    setIntermediate(false);
    if (onChange) onChange(checkedValue);
  };

  const IconName = (props.intermediate) ? 'remove' : ((checked) ? 'check' : '');
  const IconSize = (size) === 'tiny' ? 8 : 16;

  return (
    <div className={CheckboxClass}>
      <input
        type="checkbox"
        checked={checked}
        disabled={disabled}
        ref={ref}
        name={name}
        value={value}
        className={'Checkbox-input'}
      />
      <span onClick={onChangeHandler} className={CheckboxWrapper}>
        {(IconName) && <Icon name={IconName} size={IconSize} appearance={'white'} />}
      </span>
      {label && <Text small={size === 'tiny'}>{label}</Text>}
    </div>
  );
});

Checkbox.displayName = 'Checkbox';

export default Checkbox;
