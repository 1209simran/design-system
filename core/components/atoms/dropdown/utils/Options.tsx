export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
