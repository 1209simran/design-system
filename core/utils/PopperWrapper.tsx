import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
