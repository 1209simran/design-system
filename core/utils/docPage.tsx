// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
