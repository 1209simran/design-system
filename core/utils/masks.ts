import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
