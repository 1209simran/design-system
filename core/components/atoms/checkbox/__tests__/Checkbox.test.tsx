import * as React from 'react';
import { shallow } from 'enzyme';
import Checkbox, { ICheckboxProps as IProps } from '../Checkbox';
import { TestHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/TestHelper';

const Size = ['tiny', 'regular'];
const BooleanValue = [true, false];
const StringValue = ['Checkbox'];

const FunctionValue = [jest.fn()];

const Mapper: Record<string, any> = {
  size: valueHelper(Size, { iterate: true }),
  checked: valueHelper(BooleanValue, { iterate: true }),
  intermediate: valueHelper(BooleanValue, { iterate: true }),
  disabled: valueHelper(BooleanValue, { iterate: true }),
  label: valueHelper(StringValue, { iterate: true }),
  name: valueHelper(StringValue, { iterate: true }),
  value: valueHelper(StringValue, { iterate: true }),
  onChange: valueHelper(FunctionValue, { iterate: true }),
};

describe('Checkbox component', () => {
  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Checkbox
          {...attr}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  };

  TestHelper(Mapper, testFunc);
});
