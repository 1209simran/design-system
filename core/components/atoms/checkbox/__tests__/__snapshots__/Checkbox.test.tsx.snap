// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component 
  checked: false, disabled: false, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--regular"
  onClick={[Function]}
>
  <input
    checked={false}
    className="Checkbox-input"
    disabled={false}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular"
  />
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  checked: false, disabled: true, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--disabled Checkbox--regular"
  onClick={[Function]}
>
  <input
    checked={false}
    className="Checkbox-input"
    disabled={true}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular"
  />
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  checked: true, disabled: false, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--regular"
  onClick={[Function]}
>
  <input
    checked={true}
    className="Checkbox-input"
    disabled={false}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular Checkbox-wrapper--checked"
  >
    <Icon
      appearance="white"
      name="check"
      size={16}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  checked: true, disabled: true, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--disabled Checkbox--regular"
  onClick={[Function]}
>
  <input
    checked={true}
    className="Checkbox-input"
    disabled={true}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular Checkbox-wrapper--checked"
  >
    <Icon
      appearance="white"
      name="check"
      size={16}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  indeterminate: true, disabled: false, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--regular"
  onClick={[Function]}
>
  <input
    className="Checkbox-input"
    disabled={false}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular Checkbox-wrapper--indeterminate"
  >
    <Icon
      appearance="white"
      name="remove"
      size={16}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  indeterminate: true, disabled: true, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--disabled Checkbox--regular"
  onClick={[Function]}
>
  <input
    className="Checkbox-input"
    disabled={true}
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular Checkbox-wrapper--indeterminate"
  >
    <Icon
      appearance="white"
      name="remove"
      size={16}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  size: "regular", checked: true, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--regular"
  onClick={[Function]}
>
  <input
    checked={true}
    className="Checkbox-input"
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--regular Checkbox-wrapper--checked"
  >
    <Icon
      appearance="white"
      name="check"
      size={16}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={false}
    >
      Checkbox
    </Text>
  </div>
</div>
`;

exports[`Checkbox component 
  size: "tiny", checked: true, label: "Checkbox", name: "Checkbox", value: "Checkbox", onChange: "[Function]"
 1`] = `
<div
  className="Checkbox Checkbox--tiny"
  onClick={[Function]}
>
  <input
    checked={true}
    className="Checkbox-input"
    name="Checkbox"
    tabIndex={0}
    type="checkbox"
    value="Checkbox"
  />
  <span
    className="Checkbox-wrapper Checkbox-wrapper--tiny Checkbox-wrapper--checked"
  >
    <Icon
      appearance="white"
      name="check"
      size={8}
    />
  </span>
  <div
    className="Checkbox-text"
  >
    <Text
      small={true}
    >
      Checkbox
    </Text>
  </div>
</div>
`;
